packages/loki/src/LokiJsConnection.ts

```diff
@@ -2,24 +2,21 @@ import { IDatabaseConnection } from "@js-soft/docdb-access-abstractions";
 import Lokijs from "lokijs";
 import { ILokiJsDatabaseFactory } from "./ILokiJsDatabaseFactory";
 import { LokiJsCollectionProvider } from "./LokiJsCollectionProvider";
+import { LokiJsOptions } from "./LokiJsOptions";
 
 export class LokiJsConnection implements IDatabaseConnection {
     private readonly providers: Map<string, LokiJsCollectionProvider>;
 
     private static readonly defaultDatabaseFactory = {
-        create: (
-            filename: string,
-            options:
-                | (Partial<LokiConstructorOptions> & Partial<LokiConfigOptions> & Partial<ThrottledSaveDrainOptions>)
-                | undefined
-        ): Loki => {
+        create: (filename: string, options?: LokiJsOptions): Loki => {
             return new Lokijs(filename, options);
         }
     };
 
     public constructor(
         private readonly folder: string,
-        private readonly databaseFactory: ILokiJsDatabaseFactory = LokiJsConnection.defaultDatabaseFactory
+        private readonly databaseFactory: ILokiJsDatabaseFactory = LokiJsConnection.defaultDatabaseFactory,
+        private readonly lokiJsOptions: LokiJsOptions = {}
```

Comment:
Wäre es nicht logischer, die Optionen als zweites zu übergeben, und danach erst die Factory? Weil
1. die Options werden dann ja wieder in die Factory übergeben
2. man wird wahrscheinlich öfter die Options übergeben als die Factory

Reply:
Ich wollte es abwärtskompatibel halten

Reply:
Ist doch egal, Wir sind ja die einzigen dies verwenden. Kannst von mir aus auch nen Major Versionssprung machen. Aber ist doch besser als bis in alle Zukunft `new LokiJsConnection("db", undefined, options)` aufrufen zu müssen.

Reply:
Was mir außerdem gerade gekommen ist: wir könnten zusätzlich noch 2 Factory-Methoden anbieten:

```ts
LokiJsConnection#persistent(folder: string, lokiJsOptions: LokiJsOptions = {}, databaseFactory: ILokiJsDatabaseFactory = LokiJsConnection.defaultDatabaseFactory)
```

und

```ts
LokiJsConnection#inMemory(lokiJsOptions: LokiJsOptions)
```